working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
-function or method when it occurs in an augmenting function or method
-declaration. (In the latter case, the augmenting method declaration must
-occur inside an augmenting type-introducing declaration, e.g., an
-augmenting class or mixin declaration). This is also true if `augmented()`
-occurs inside a local function declaration inside the body of that function
-or method declaration. We could say that `augmented` is a contextual
-reserved word because it is unable to refer to a declaration in scope when
-it occurs inside an augmenting declaration, it always has the special
-meaning which is associated with augmentations.*
-
-A compile-time error occurs if a declaration with the basename `augmented`
-occurs in a location where any enclosing declaration is augmenting. *This
-error is applicable to all such declarations, e.g., local functions, local
-variables, parameters, and type parameters.*
-
-Consider a non-augmenting member declaration _Dm_ that occurs inside an
-augmenting type declaration _Dt_. A compile-time error occurs if the
-identifier `augmented` occurs in _Dm_.
-
-*For example, inside `augment class C` we could have a declaration like
-`void f() {...augmented()...}`. This is an error because the outer
-`augment` forces the meaning of `augmented` to be about augmentation in the
-entire scope, but the method declaration is introductory and hence there is
-no earlier declaration to augment.*
+invoked with the same value for `this`. For a non-redirecting factory
+constructor, it's invoked with the same type parameter bindings for the
+surrounding type.
+
+The grammar change for `augmented` adds the following productions:
+```ebnf
+  <augmentedValueExpression> ::= 'augmented'
+  <augmentedSetterExpression> ::= 'augmented' '=' <expression>
+  <augmentedFunctionExpression> ::= 'augmented' <argumentPart>
+  <augmentedConstructorExpression> ::= 'augmented' <argumentList>
+
+  <primary> ::= ...
+     | <augmentedValueExpression>        -- augmenting variable or getter body
+     | <augmentedSetterExpression>       -- augmenting setter body
+     | <augmentingFunctionExpression>    -- augmenting function, operator body
+     | <augmentingConstructorExpression> -- augmenting factory constructor body
+```
+and makes `augmented` a reserved word inside any expression or body of an
+augmenting declaration in the list above. The grammar production is only
+allowed inside the scopes mentioned. _This makes occurrences in parameter
+default values and initializer list expressions invalid._
+
+_The word is also reserved, and has its designated meaning, inside the bodies of
+nested functions inside the expressions or bodies of the augmenting declarations.
+It's possible to close over `augmented` inside an augmenting function, to,
+for example, do `list.forEach((x) => augmented(x));` in a function augmentation,
+but doing `List.forEach(augmented)` is not allowed. That would treat the
+augmented function as if it had an identity._
```

Comment:
This is a good thing to point out. It feels weird at first, but then I realized that `super` has the same limitation and it's never bothered me.